Assertion error when using existing `log_group` for `awsx.ecs.FargateTaskDefinition`

### What happened?

Using Python I wanted to create a `awsx.ecs.FargateTaskDefinition` that used a previously created log group. I tried to use the `log_group` parameter for this, which has the type `Optional[pulumi.InputType['_awsx.DefaultLogGroupArgs']]`. 

### Steps to reproduce

A `awsx.ecs.FargateTaskDefinition` instance was created with the log group set as follows:

```python
awsx.ecs.FargateTaskDefinition(
    resource_name=f"{resource_key}-orchestrator-task",
    log_group=awsx.awsx.DefaultLogGroupArgs(
        existing=awsx.awsx.ExistingLogGroupArgs(arn=log_group.arn)
    ),
    container=awsx.ecs.TaskDefinitionContainerDefinitionArgs(
        command=command,
        image=image.image_uri,
        cpu=256,
        memory=512,
        essential=True,
    ),
)
```

Where `log_group.arn` is the ARN of a previously created log group.

### Expected Behavior

The previously created log group should be used by tasks created by the task definition. No errors should occur during `pulumi up`.

### Actual Behavior

The following error appears in the diagnostics section after running `pulumi up`.

```
    error: Program failed with an unhandled exception:
    Traceback (most recent call last):
      File "/home/will/.pulumi/bin/pulumi-language-python-exec", line 179, in <module>
        loop.run_until_complete(coro)
      File "/home/will/.pyenv/versions/3.10.7/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
        return future.result()
      File "/home/will/.pyenv/versions/dragon/lib/python3.10/site-packages/pulumi/runtime/stack.py", line 126, in run_in_stack
        await run_pulumi_func(lambda: Stack(func))
      File "/home/will/.pyenv/versions/dragon/lib/python3.10/site-packages/pulumi/runtime/stack.py", line 51, in run_pulumi_func
        await wait_for_rpcs()
      File "/home/will/.pyenv/versions/dragon/lib/python3.10/site-packages/pulumi/runtime/stack.py", line 110, in wait_for_rpcs
        raise exception
      File "/home/will/.pyenv/versions/dragon/lib/python3.10/site-packages/pulumi/output.py", line 168, in run
        is_known = await self._is_known
      File "/home/will/.pyenv/versions/dragon/lib/python3.10/site-packages/pulumi/output.py", line 99, in is_value_known
        return await is_known and not contains_unknowns(await future)
      File "/home/will/.pyenv/versions/dragon/lib/python3.10/site-packages/pulumi/runtime/resource.py", line 653, in do_register
        rpc.resolve_outputs(
      File "/home/will/.pyenv/versions/dragon/lib/python3.10/site-packages/pulumi/runtime/rpc.py", line 1049, in resolve_outputs
        all_properties[translated_key] = translate_output_properties(
      File "/home/will/.pyenv/versions/dragon/lib/python3.10/site-packages/pulumi/runtime/rpc.py", line 882, in translate_output_properties
        raise AssertionError(
    AssertionError: Unexpected type; expected a value of type `<class 'pulumi_aws.cloudwatch.log_group.LogGroup'>` but got a value of type `<class 'dict'>` at resource `ABCD-orchestrator-task`, property `log_group`: {'existing': {'arn': <pulumi.output.Output object at 0x7f5bc865b460>}}
```

### Output of `pulumi about`
```
CLI          
Version      3.42.0
Go Version   go1.19.1
Go Compiler  gc

Plugins
NAME     VERSION
aws      5.16.0
awsx     1.0.0-beta.11
docker   3.4.1
python   unknown

Host     
OS       linuxmint
Version  20.3
Arch     x86_64

This project is written in python: executable='/home/will/.pyenv/shims/python3' version='3.10.7
'

Current Stack: dev

Found no resources associated with dev

Found no pending operations associated with dev

Backend        
Name           desktop
URL            file://~
User           will
Organizations  

Pulumi locates its logs in /tmp by default
```

### Additional context

The documentation for the `log_group` parameter of `FargateTaskDefinition` is:

> A set of volume blocks that containers in your task may use.

Which seems wrong.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,7 @@ CHANGELOG
   provide a top-level way to set the default target group's port.
 * Change the behavior of `ecs.FargateTaskDefinition` and `ecs.EC2TaskDefinition` to always respect the passed-in
   `hostPort`, regardless of the target group's port.
+* Python: Bump the min required version of `pulumi` PyPi package to 3.47.2.
 
 ## 0.40.0 (2022-03-24)
 * Compatibility with pulumi-aws v5.0.0

--- a/awsx/schema.json
+++ b/awsx/schema.json
@@ -2895,7 +2895,7 @@
             "liftSingleValueMethodReturns": true,
             "readme": "Pulumi Amazon Web Services (AWS) AWSX Components.",
             "requires": {
-                "pulumi": "\u003e=3.0.0,\u003c4.0.0",
+                "pulumi": "\u003e=3.47.2,\u003c4.0.0",
                 "pulumi-aws": "\u003e=5.3.0,\u003c6.0.0",
                 "pulumi-docker": "\u003e=3.0.0,\u003c4.0.0"
             },

--- a/schemagen/pkg/gen/schema.go
+++ b/schemagen/pkg/gen/schema.go
@@ -86,7 +86,7 @@ func GenerateSchema(packageDir string) schema.PackageSpec {
 			}),
 			"python": rawMessage(map[string]interface{}{
 				"requires": map[string]string{
-					"pulumi":        ">=3.0.0,<4.0.0",
+					"pulumi":        ">=3.47.2,<4.0.0",
 					"pulumi-aws":    ">=5.3.0,<6.0.0",
 					"pulumi-docker": ">=3.0.0,<4.0.0",
 				},

--- a/sdk/python/setup.py
+++ b/sdk/python/setup.py
@@ -60,7 +60,7 @@ setup(name='pulumi_awsx',
       },
       install_requires=[
           'parver>=0.2.1',
-          'pulumi>=3.0.0,<4.0.0',
+          'pulumi>=3.47.2,<4.0.0',
           'pulumi-aws>=5.3.0,<6.0.0',
           'pulumi-docker>=3.0.0,<4.0.0',
           'semver>=2.8.1'

